src/app/components/reports/pages/event-summary/EventSummaryReportParameters.ts

```diff
@@ -0,0 +1,174 @@
+      timeStartedAfter: (value: string) => Duration.fromObject({
+        hours: Number(value.split(":")[0]),
+        minutes: Number(value.split(":")[1]),
+      }),
+      timeFinishedBefore: (value: string) => Duration.fromObject({
+        hours: Number(value.split(":")[0]),
+        minutes: Number(value.split(":")[1]),
+      }),
+    };
+
+    // since query string parameters are losely typed using a string from the user space
+    // we have to use this hacky implementation of manual key-value checking and assignment
+    Object.entries(queryStringParameters).forEach(([key, value]) => {
+      if (key in conversionTable) {
+        this[key] = conversionTable[key](value);
+      }
+    });
+  }
+
+  // since these properties are exposed to the user in the form of query string parameters
+  // we use the user friendly names
+  public sites: Id[];
+  public points: Id[];
+  public provenances: Id[];
+  public events: Id[];
+  public recogniserCutOff = 0;
+  public charts: string[];
+  public binSize: BinSize = BinSize.month;
+  public ignoreDaylightSavings = true;
+  public timeStartedAfter: Duration;
+  public timeFinishedBefore: Duration;
+  public dateStartedAfter: DateTime;
+  public dateFinishedBefore: DateTime;
+
+  public toFilter(): Filters<EventSummaryReport> {
+    let filter: InnerFilter<EventSummaryReport>;
+
+    if (this.sites) {
+      filter = filterModelIds<EventSummaryReport>("region", this.sites, filter);
+    }
+
+    if (this.points) {
+      filter = filterModelIds<EventSummaryReport>("site", this.points, filter);
+    }
+
+    if (this.provenances) {
+      filter = filterModelIds<EventSummaryReport>(
+        "provenance",
+        this.provenances,
+        filter
+      );
+    }
+
+    if (this.events) {
+      filter = filterModelIds<EventSummaryReport>("tag", this.events, filter);
+    }
+
+    // we use isInstantiated() here because 0 is a valid value for recogniserCutOff
+    if (isInstantiated(this.recogniserCutOff)) {
+      filter = filterAnd<EventSummaryReport>(filter, {
+        score: {
+          gteq: this.recogniserCutOff,
+        },
+      } as InnerFilter);
+    }
+
+    if (this.binSize) {
+      filter = filterAnd<EventSummaryReport>(filter, {
+        binSize: {
+          eq: this.binSize,
+        },
+      } as InnerFilter);
+    }
+
+    if (this.dateStartedAfter || this.dateFinishedBefore) {
+      filter = filterDate(
+        filter,
+        this.dateStartedAfter,
+        this.dateFinishedBefore
+      );
+    }
+
+    if (this.timeStartedAfter || this.timeFinishedBefore) {
+      filter = filterTime(
+        filter,
+        this.ignoreDaylightSavings,
+        this.timeStartedAfter,
+        this.timeFinishedBefore
+      );
+    }
+
+    return { filter };
+  }
+
+  /** Converts the report parameters to filters and base64 encodes them */
+  public toFilterString(): string {
+    return toBase64Url(JSON.stringify(this.toFilter()));
+  }
+
+  public toQueryString(): string {
```

Comment:
If we're using a `StrongRoute`, we should use the `[queryParameters]` input

We can then theoretically pass the model directly to the strong route and let it handle QSP conversion.

I don't know how I feel about this though, because the serialization and deserialization would be using two different functions